File: db4o.net/Db4oBench/Bench/Util/IoBenchmarkArgumentParser.cs
/* Copyright (C) 2004 - 2008  Versant Inc.  http://www.db4o.com */

using Sharpen;

namespace Db4objects.Db4o.Bench.Util
{
	public class IoBenchmarkArgumentParser
	{
		private string _resultsFile2;

		private string _resultsFile1;

		private int _objectCount;

		private bool _delayed;

		public IoBenchmarkArgumentParser(string[] arguments)
		{
			ValidateArguments(arguments);
		}

		private void ValidateArguments(string[] arguments)
		{
			if (arguments.Length != 1 && arguments.Length != 3)
			{
				Sharpen.Runtime.Out.WriteLine("Usage: IoBenchmark <object-count> [<results-file-1> <results-file-2>]"
					);
                throw new System.Exception("Usage: IoBenchmark <object-count> [<results-file-1> <results-file-2>]");
			}
			_objectCount = int.Parse(arguments[0]);
			if (arguments.Length > 1)
			{
				_resultsFile1 = arguments[1];
				_resultsFile2 = arguments[2];
				_delayed = true;
			}
		}

		public virtual int ObjectCount()
		{
			return _objectCount;
		}

		public virtual string ResultsFile1()
		{
			return _resultsFile1;
		}

		public virtual string ResultsFile2()
		{
			return _resultsFile2;
		}

		public virtual bool Delayed()
		{
			return _delayed;
		}
	}
}

File: db4otools/src/com/db4o/bench/util/IoBenchmarkArgumentParser.java
/* Copyright (C) 2004 - 2007 Versant Inc. http://www.db4o.com */

package com.db4o.bench.util;

/**
 * 
 * @sharpen.ignore
 *
 */
public class IoBenchmarkArgumentParser {
	
	private String _resultsFile2;
	private String _resultsFile1;
	private int _objectCount;
	private boolean _delayed;

	public IoBenchmarkArgumentParser(String[] arguments) {
		validateArguments(arguments);
	}
		
	private void validateArguments(String[] arguments) {
		if (arguments.length != 1 && arguments.length != 3) {
			System.out.println("Usage: IoBenchmark <object-count> [<results-file-1> <results-file-2>]");
			throw new RuntimeException("Usage: IoBenchmark <object-count> [<results-file-1> <results-file-2>]");
		}
		_objectCount = java.lang.Integer.parseInt(arguments[0]);
		if (arguments.length > 1) {
			_resultsFile1 = arguments[1];
			_resultsFile2 = arguments[2];
			_delayed = true;
		}
	}


	public int objectCount() {
		return _objectCount;
	}
	
	public String resultsFile1() {
		return _resultsFile1;
	}
	
	public String resultsFile2() {
		return _resultsFile2;
	}
	
	public boolean delayed() {
		return _delayed;
	}
}
